Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/ToolbarController.java

```diff
@@ -193,6 +196,20 @@ public void setDisabledAnimation(boolean disabled) {
         animate.setDisabled(disabled || !userOptions.getMainAttributeKey().equals(parent.getConfigData().getDefaultAttribute()));
     }
 
+    public void setDisabledCost(boolean disabled) {
+        boolean hasNoRoles = false;
+        try {
+            ImmutableList<Object> roleValues = parent.getProcessAnalyst().getRoleValues();
+            if (roleValues.isEmpty()) {
+                hasNoRoles = true;
+            }
+        } catch (Exception e) {
+            hasNoRoles = true;
+        }

```

Comment:
There could be other exception thrown out in case of errors but has been silenced. The PDAnalyst.getRoleValues should handle null like what has been done in https://github.com/apromore/ApromoreCore/pull/2301.